minima/database.nim

```diff
@@ -0,0 +1,91 @@
+# Minima
+#
+# Copyright (c) 2020 Dean Eigenmann
+# Licensed under MIT license ([LICENSE](LICENSE) or http://opensource.org/licenses/MIT)
+
+## This module implements Minima's basic key value database.
+
+import stew/results, os, io, tree
+
+type 
+    ## Database object
+    Database* = ref object
```

Comment:
Not sure about the use of ref here, in most cases, unless you are planning on using dynamic dispatch you can get away with stack object. I guess it's fine for now, but worth re-evaluating at some point.